server/utils/passport/ensure-authenticated.js

```diff
@@ -13,12 +13,18 @@ export default async function ensureAuthenticated(req, res, next) {
 
     switch (true) {
       case isAccountExpired(account_expires, role):
-        await req.session.destroy()
-        await req.logout()
-
         res.clearCookie('connect.sid')
 
+        req.logout(function (err) {
+          if (err) return next(err)
+
+          req.session.destroy(function (sessionError) {
+            if (sessionError) return next(sessionError)
+          })
+        })
```

Comment:
we run this code block in 3 different files, which makes me think we should extract it and DRY it up.

Reply:
[b194a00](https://github.com/AMP-SCZ/dpdash/pull/579/commits/b194a003ed6a3e4d458303341b26bfb2c79c7f88)